File: 0215-kth-largest-element-in-an-array/0215-kth-largest-element-in-an-array.cpp
class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        priority_queue<int>heap; // default is maxHeap : 
        for(int num : nums){
            heap.push(num);
        }
         k--;
       while(k--) {
            int top = heap.top();
             heap.pop();
          }
  return heap.top();
    }
};

File: 0215-kth-largest-element-in-an-array/0215-kth-largest-element-in-an-array.java
class Solution {
    public int findKthLargest(int[] nums, int k) {
       PriorityQueue<Integer>pq = new PriorityQueue<>();  
       //  minHeap
      // PriorityQueue<Integer>pq = new PriorityQueue<>((a,b) -> a > b ? a : b);
        for(int num : nums){
        pq.add(num);
             if(pq.size() > k){
                pq.poll();
             }
           System.out.println(pq.peek());  
        }    
      return pq.poll();
    }
}